### ISSUE TITLE
Unable to build for Android with version 1.1.4+

### ISSUE THREAD

**Issue by peternormann** [natmlx/natml-unity#54]
I have been trying to build a unity project for Android, but whenever I use version 1.1.3, 1.1.4 or 1.1.5 the build will fail

```
DllNotFoundException: NatML assembly:<unknown assembly> type:<unknown type> member:(null)
NatML.MLEdgeModel.CreateSecret () (at ./Library/PackageCache/ai.natml.natml@1.1.4/Runtime/MLEdgeModel.cs:292)
NatML.Editor.NatMLSettingsEmbed.CreateEmbeds (UnityEditor.Build.Reporting.BuildReport report) (at ./Library/PackageCache/ai.natml.natml@1.1.4/Editor/NatMLSettingsEmbed.cs:50)
NatML.Hub.Editor.BuildEmbedHelper`1[T].UnityEditor.Build.IPreprocessBuildWithReport.OnPreprocessBuild (UnityEditor.Build.Reporting.BuildReport report) (at ./Library/PackageCache/ai.natml.hub@1.0.20/Editor/BuildEmbedHelper.cs:89)
UnityEditor.Build.BuildPipelineInterfaces+<>c__DisplayClass16_0.<OnBuildPreProcess>b__1 (UnityEditor.Build.IPreprocessBuildWithReport bpp) (at /home/bokken/build/output/unity/unity/Editor/Mono/BuildPipeline/BuildPipelineInterfaces.cs:499)
UnityEditor.Build.BuildPipelineInterfaces.InvokeCallbackInterfacesPair[T1,T2] (System.Collections.Generic.List`1[T] oneInterfaces, System.Action`1[T] invocationOne, System.Collections.Generic.List`1[T] twoInterfaces, System.Action`1[T] invocationTwo, System.Boolean exitOnFailure) (at /home/bokken/build/output/unity/unity/Editor/Mono/BuildPipeline/BuildPipelineInterfaces.cs:465)
UnityEditor.BuildPlayerWindow:BuildPlayerAndRun() (at /home/bokken/build/output/unity/unity/Editor/Mono/BuildPlayerWindow.cs:188)
```

I have tried building with both Mono and IL2CP and with several editor versions in the range 2021.3-2023.2.

I am building on linux.

I am new to Unity, so I might be doing something wrong. I am hoping you can give me some pointers and if you need me to dig out more information, I'll be happy to help.

Also, I noted there seem to be some issues with the inter-dependencies between packages that also seem to nag. Some packages, e.g. yolox-unity, depend on earlier versions or have dependencies that do. In the unlikely case it's not me misunderstanding something fundamental, this will cause problems using the latest versions.

**Comment by olokobayusuf** [comment]
Hey @peternormann, you're getting this error because we don't yet have support for Linux. Given that you are only building on Linux (and not trying to run on the platform), we can add a workaround to get you past this error. I should be able to get an update with the workaround in a day or two.

> Also, I noted there seem to be some issues with the inter-dependencies between packages that also seem to nag. Some packages, e.g. yolox-unity, depend on earlier versions or have dependencies that do. In the unlikely case it's not me misunderstanding something fundamental, this will cause problems using the latest versions.

Please report all of these! We've made some API changes which might cause compiler errors. We've been updating the predictor samples, but haven't yet gotten to all of them.

**Comment by olokobayusuf** [comment]
[Fixed in 1.1.6](https://github.com/natmlx/natml-unity/releases/tag/1.1.6).
